src/encoded/static/components/filegallery.js

```diff
@@ -528,24 +549,61 @@ class RawSequencingTable extends React.Component {
                                 file.biological_replicates[0] === partner.biological_replicates[0]) ||
                                 ((!file.biological_replicates || file.biological_replicates.length === 0) &&
                                 (!partner.biological_replicates || partner.biological_replicates.length === 0))) {
-                            // Both the file and its partner qualify as good pairs of each other. Let
-                            // them pass the filter, and record set their sort keys to the lower of
-                            // the two accessions -- that's how pairs will sort within a biological
-                            // replicate.
-                            partner.pairSortKey = file.title < partner.title ? file.title : partner.title;
-                            file.pairSortKey = partner.pairSortKey;
-                            file.pairSortKey += file.paired_end;
-                            partner.pairSortKey += partner.paired_end;
+                            // Both the file and its partner qualify as good pairs of each other.
+                            // Let hem pass the filter, and set their sort keys to the lower of the
+                            // two accessions -- that's how pairs will sort within a biological
+                            // replicate. Also track the `pairSortId` which is the same as
+                            // `portSortKey` but without the paired_end property concatenated.
+                            partner.pairSortId = file.title < partner.title ? file.title : partner.title;
+                            partner.pairSortKey = `${partner.pairSortId}-${partner.paired_end}`;
+                            file.pairSortId = partner.pairSortId;
+                            file.pairSortKey = `${partner.pairSortId}-${file.paired_end}`;
                             return true;
                         }
                     }
                 }
 
-                // File not part of a pair; add to non-paired list and filter it out
-                nonpairedFiles.push(file);
+                // File not part of a pair; add it to the non-paired list as a copy so we can
+                // mutate them later.
+                nonpairedFiles.push(Object.assign({}, file));
```

Comment:
Is there a created a new object  via Object.assign. rather than use file-object

It would be cool if we upgrade babel to support object spread (https://stackoverflow.com/q/32925460/178550)

Reply:
I’m mutating the file objects, and the file objects come in as a prop.